core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -394,16 +410,22 @@ Builder specsById(Map<Integer, PartitionSpec> newSpecsById) {
     }
 
     Builder filterData(Expression newDataFilter) {
+      Preconditions.checkArgument(
```

Comment:
If there is a use case in the future where we need to support these filters, we can add implement these methods. That said, it won't be needed for distributed planning and it won't be trivial.